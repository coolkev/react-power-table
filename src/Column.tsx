import * as React from 'react';
import { makePure, debuglog } from "./utils";
import { HeaderComponentProps, CellProps, Column, GridProps, StrictColumn } from "./ReactPowerTable";


const defaultHeaderComponent = makePure((props: HeaderComponentProps) => <div>{props.headerText}</div>);
/** @internal */
export const defaultCellComponent = makePure((props: CellProps<any>) => {
    debuglog('Render cellComponent column: ' + props.column.key + ' value: ' + props.value);
    return <div>{props.value}</div>;
});

/** @internal */
export function transformColumn<T>(options: Column<T> | string, gridProps: GridProps<T>): StrictColumn<T> {

    debuglog('transformColumn', options);

    if (typeof options === 'string') {
        options = { field: options } as Column<T>;
    }

    //const { key, field } = getKeyAndField(options);
    const core = getColumnCore(options);
    const { cellProps, headerProps } = getCellAndHeaderProps(options);


    const result: StrictColumn<T> = {
        //__transformed: true,
        ...options,
        ...core,
        cellProps: cellProps,
        headerCellProps: headerProps,
        formatter: options.formatter || null
    };

    if (options.cellComponent) {
        result.cellComponent = options.cellComponent;
        result.cellComponentProps = options.cellComponentProps || (props => props);

    }
    else {

        result.cellComponent = defaultCellComponent;
        result.cellComponentProps = options.cellComponentProps || (props => ({ column: props.column, value: props.value }));

    }


    result.headerComponent = options.headerComponent || gridProps.defaultHeaderComponent || defaultHeaderComponent;
    //result.headerComponentPropsProvider = options.headerComponentPropsProvider || (()=> ({headerText:result.headerText, key: result.key , headerCellProps: result.headerCellProps}));

    const originalHeaderComponentPropsProvider = options.headerComponentPropsProvider;

    result.headerComponentPropsProvider = () => {

        const originalProps = originalHeaderComponentPropsProvider && originalHeaderComponentPropsProvider();
        
        return { ...originalProps, headerText:result.headerText, key: result.key , headerCellProps: result.headerCellProps };
    }

    return result;
}




function getCellAndHeaderProps(options: Column<any>) {

    const cssClass = options.cssClass;

    const headerProps = { ...options.headerCellProps };
    let cellProps: ((props: CellProps<any>) => React.HTMLProps<HTMLTableDataCellElement>);

    const cellStaticProps: React.HTMLProps<HTMLTableDataCellElement> = typeof (options.cellProps) === 'function' ? {} : { ...options.cellProps };
    const cellPropsFunc = typeof (options.cellProps) === 'function' ? options.cellProps : () => null;


    //var cssClassFunc: (row: T) => string;

    if (options.width) {
        cellStaticProps.style = { ...cellStaticProps.style, width: options.width };
        headerProps.style = { ...headerProps.style, width: options.width };

    }

    if (options.maxWidth) {
        cellStaticProps.style = { ...cellStaticProps.style, maxWidth: options.maxWidth };
        headerProps.style = { ...headerProps.style, maxWidth: options.maxWidth };

    }

    if (typeof cssClass === 'function') {
        cellProps = row => ({ ...cellStaticProps, ...cellPropsFunc(row), className: cssClass(row) });

    }
    else if (typeof (cssClass) === "string") {
        cellStaticProps.className = cssClass;
        cellProps = row => ({ ...cellStaticProps, ...cellPropsFunc(row) });
    }
    else {
        cellProps = row => ({ ...cellStaticProps, ...cellPropsFunc(row) });
    }
    if (options.headerCssClass) {
        headerProps.className = options.headerCssClass;
    }

    if (options.textAlign) {
        cellStaticProps.style = { ...cellStaticProps.style, textAlign: options.textAlign };
        headerProps.style = { ...headerProps.style, textAlign: options.textAlign };
    }

    debuglog('getCellAndHeaderProps', options);
    return { cellProps, headerProps };

}


export interface ColumnCore<T> {
    key: string | number;
    field: (row: T) => any;
    fieldName: string;
    headerText: string;
}

export function getColumnCore<T>(col: Column<T> | string): ColumnCore<T> {

    //let fieldName: string;
    //let field: (row) => any;

    if (typeof (col) == 'string') {
        return {
            key: col,
            field: row => row[col],
            fieldName: col,
            headerText: col,
        
        };
    }

    const { field, key } = col;
    if (typeof field === 'function') {
        const fieldName = getExpression(field);
        return {
            key: key || fieldName,
            field: field,
            fieldName: fieldName,
            headerText: col.headerText || fieldName
        };
    }
    if (typeof (field) === "string") {
        return {
            key: key || field,
            field: row => row[field],
            fieldName: field,
            headerText: col.headerText || field

        };
    }

    if (!key) {
        throw new Error('Must specify value for "key" if field is not used')
    }
    //    field = _row => null;

    return {
        key: key,
        field: _row => null,
        fieldName: null,
        headerText: col.headerText
        
    };

}
/** @internal */
export function getExpression(func: Function): string {

    const expr = func.toString();

    let myregexp = /(?:return|\w+ => ).*\.(\w+);?/;
    let match = myregexp.exec(expr);
    if (match != null) {
        return match[1];
    }
    return null;
}
