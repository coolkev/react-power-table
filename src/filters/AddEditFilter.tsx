import * as React from 'react';
import Radio from 'react-bootstrap/lib/Radio';
import Button from 'react-bootstrap/lib/Button';
import { debuglog } from "../utils";
import * as PowerTable from "./definitions/FilterDefinition";

export interface AddEditFilterProps {
    filter: PowerTable.FilterDefinition<any>;
    initialOperation: PowerTable.OperationDefinition<any>;
    initialValue?: any;
    onApplyFilter: (filter: PowerTable.AppliedFilter<any>) => void;
}

export interface AddEditFilterState {

    operation: PowerTable.OperationDefinition<any>;
    value: any;

}


export class AddEditFilter extends React.PureComponent<AddEditFilterProps, AddEditFilterState> {

    constructor(props: AddEditFilterProps) {
        super(props);


        const value = props.initialValue || props.filter.defaultValue || '';

        this.state = { value: value, operation: props.initialOperation };

        this.selectedOperationChange = this.selectedOperationChange.bind(this);
        this.handleFilterValueChange = this.handleFilterValueChange.bind(this);
        this.applyFilter = this.applyFilter.bind(this);

    }

    private selectedOperationChange(e: React.FormEvent<Radio & HTMLInputElement>) {

        const prevOperation = this.state.operation;
        const operationKey = e.currentTarget.value;
        const operation = this.props.filter.operations[operationKey];

        const newState: Partial<AddEditFilterState> = { operation: operation };

        if (prevOperation.key == 'between' && operation.key != prevOperation.key && typeof this.state.value === 'string') {
            newState.value = this.state.value.split(' ')[0];
        }
        this.setState(newState as any)
    }

    private handleFilterValueChange(value: string) {
        debuglog('AddEditFilter.filterValueChange', value);
        this.setState({
            value: value
        });
    }

    private applyFilter() {
        this.props.onApplyFilter({ filter: this.props.filter, operation: this.state.operation, value: this.state.value });
    }
    render() {
        debuglog('AddEditFilter.render', this.props);

        const filter = this.props.filter;


        return <div>
            <div style={{ margin: '10px 0' }}><b>{filter.displayName}</b>
            </div>




            {filter.operations.all.map(op => {
                
                let selectedFilterComponentProps: PowerTable.FilterComponentProps<any>;
                let SelectedFilterComponent: React.ComponentClass<PowerTable.FilterComponentProps<any>> | React.StatelessComponent<PowerTable.FilterComponentProps<any>>;

                if (op.key == this.state.operation.key) {
                    selectedFilterComponentProps = {
                        filter: this.props.filter,
                        operation:op,
                        value: this.state.value,
                        onValueChange: this.handleFilterValueChange,
                        onEnterKeyPress: this.applyFilter
                    };

                    SelectedFilterComponent = op.filterComponent || filter.filterComponent;

                }



                return <div key={op.key}>
                    <Radio checked={op.key == this.state.operation.key} onChange={this.selectedOperationChange} value={op.key}>
                        {filter.radioButtonLabel ? filter.radioButtonLabel({
                            filter: this.props.filter,
                            operation: op
                        }) : op.displayName}

                    </Radio>
                    {SelectedFilterComponent &&
                        <div style={{ marginLeft: 20 }}><SelectedFilterComponent {...selectedFilterComponentProps} /></div>
                    }
                </div>;


            })}


            <div style={{ marginTop: 20 }}>
                <Button bsStyle="primary" bsSize="sm" onClick={this.applyFilter}>Apply Filter</Button>
            </div>

        </div>;
    }
}
