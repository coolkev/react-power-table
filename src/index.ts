export * from './ReactPowerTable';
export * from './Paging';
export * from './Sorting';
export * from './GridFilters';
export * from './filters/';
export {groupBy,sortArray, numberWithCommas} from './utils';
