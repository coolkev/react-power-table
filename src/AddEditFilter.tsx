import * as React from 'react';
import * as Button from 'react-bootstrap/lib/Button';
import * as Radio from 'react-bootstrap/lib/Radio';
import * as filters from './filters/FilterDefinition';
import { debuglog } from './utils';

/**
 * @internal
 */
export interface AddEditFilterProps {
    filter: filters.FilterDefinition<any>;
    initialOperation: filters.OperationDefinition<any>;
    initialValue?: any;
    onApplyFilter: (filter: filters.AppliedFilter<any>) => void;
}

/**
 * @internal
 */
export interface AddEditFilterState {

    operationKey: string;
    value: any;

}

/**
 * @internal
 */
export class AddEditFilter extends React.PureComponent<AddEditFilterProps, AddEditFilterState> {

    constructor(props: AddEditFilterProps) {
        super(props);

        const value = props.initialValue || props.filter.defaultValue || '';

        this.state = { value, operationKey: props.initialOperation.key };

        this.selectedOperationChange = this.selectedOperationChange.bind(this);
        this.handleFilterValueChange = this.handleFilterValueChange.bind(this);
        this.applyFilter = this.applyFilter.bind(this);

    }

    private selectedOperationChange(e: React.FormEvent<Radio & HTMLInputElement>) {

        const prevOperationKey = this.state.operationKey;
        const operationKey = e.currentTarget.value;
        //const operation = this.props.filter.operations[operationKey];

        const newState: AddEditFilterState = { operationKey, value: this.state.value };

        this.props.filter.onOperationChange(prevOperationKey, newState);
        // if (prevOperationKey === 'between' && operationKey !== prevOperationKey && typeof this.state.value === 'string') {
        //     newState.value = this.state.value.split(' ')[0];
        // }
        this.setState(newState);
    }

    private handleFilterValueChange(value: string) {
        debuglog('AddEditFilter.filterValueChange', value);
        this.setState({
            value,
        });
    }

    private applyFilter() {
        const operation = this.props.filter.operations[this.state.operationKey];

        this.props.onApplyFilter({ filter: this.props.filter, operation, value: this.state.value });
    }
    render() {
        debuglog('AddEditFilter.render', this.props);

        const filter = this.props.filter;

        const { value, operationKey } = this.state;

        const opComponents = Object.keys(filter.operations).map((opKey) => {
            const op = filter.operations[opKey];
            let children: React.ReactChild;
            if (opKey === this.state.operationKey) {

                const SelectedFilterComponent = op.filterComponent || filter.filterComponent;

                if (SelectedFilterComponent) {
                    children = <div style={{ marginLeft: 20 }}><SelectedFilterComponent filter={filter} operation={op} value={value} onValueChange={this.handleFilterValueChange} onEnterKeyPress={this.applyFilter} /></div>;
                }
            }

            return <FilterOperation key={opKey} operation={op} filter={filter} selected={opKey === operationKey} onChange={this.selectedOperationChange}>{children}</FilterOperation>;
        });

        return (
            <div>
                <div style={{ margin: '10px 0' }}><b>{filter.displayName}</b>
                </div>

                {opComponents.map((opComponent) => opComponent)}

                <div style={{ marginTop: 20 }}>
                    <Button bsStyle="primary" bsSize="sm" onClick={this.applyFilter}>Apply Filter</Button>
                </div>

            </div>);
    }
}

/**
 * @internal
 */
export interface FilterOperationProps extends React.Props<any> {
    operation: filters.OperationDefinition;
    filter: filters.FilterDefinition;

    selected: boolean;
    onChange: React.EventHandler<React.FormEvent<Radio & HTMLInputElement>>;
}

/**
 * @internal
 */
export const FilterOperation = (props: FilterOperationProps) => {

    const { operation, filter, onChange, selected, children } = props;

    return (
        <div key={operation.key}>
            <Radio checked={selected} onChange={onChange} value={operation.key}>
                {filter.radioButtonLabel ? filter.radioButtonLabel({ filter, operation }) : operation.displayName}

            </Radio>
            {children}
        </div>
    );

};
