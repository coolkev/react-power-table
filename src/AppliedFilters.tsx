import * as React from 'react';
import { ActionLink } from './components/ActionLink';
import * as filters from './filters';

export interface AppliedFiltersProps {
    appliedFilters: ReadonlyArray<filters.AppliedFilter>;
    removeFilter: (filter: filters.AppliedFilter) => void;
    editFilter: (filter: filters.AppliedFilter) => void;

}

export const AppliedFilters = (props: AppliedFiltersProps) => {
    return (
        <div className="applied-filters small">
            {props.appliedFilters.map((appliedFilter) => <AppliedFilter appliedFilter={appliedFilter} key={appliedFilter.key} onEditFilter={props.editFilter} onRemoveFilter={props.removeFilter} />)}
        </div>
    );
};

export interface AppliedFilterProps {
    appliedFilter: filters.AppliedFilter;
    onRemoveFilter: (filter: filters.AppliedFilter) => void;
    onEditFilter: (filter: filters.AppliedFilter) => void;

}

const defaultAppliedFilterLabelComponent = (props: filters.AppliedFilter<any>) => {
    const appliedLabel = props.operation.appliedLabel || props.filter.appliedLabel || filters.FilterDefinition.defaultAppliedFilterLabel;
    return <span>{appliedLabel(props)}</span>;
};

export const AppliedFilter = ({ appliedFilter, onRemoveFilter, onEditFilter }: AppliedFilterProps) => {

    const handleRemoveFilter = React.useCallback((e: React.SyntheticEvent<any>) => {
        e.preventDefault();
        onRemoveFilter(appliedFilter);
    }, [appliedFilter, onRemoveFilter]);

    const handleEditFilter = React.useCallback(() => {

        onEditFilter(appliedFilter);

    }, [appliedFilter, onEditFilter]);

    const AppliedLabelComponent = appliedFilter.operation.appliedLabelComponent || appliedFilter.filter.appliedLabelComponent || defaultAppliedFilterLabelComponent;

    return (
        <div className="applied-filter well well-sm" style={{ marginBottom: 10 }} key={appliedFilter.filter.fieldName}>
            <button type="button" className="close" aria-label="Remove" onClick={handleRemoveFilter}><span aria-hidden="true">Ã—</span></button>
            <ActionLink onClick={handleEditFilter}><AppliedLabelComponent {...appliedFilter} /></ActionLink>
        </div>
    );

};

