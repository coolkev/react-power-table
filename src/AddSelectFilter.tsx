import * as React from 'react';
import * as FormControl from 'react-bootstrap/lib/FormControl';
import { AddEditFilter } from './AddEditFilter';
import { BackLink } from './components/BackLink';
import * as PowerTable from './filters/FilterDefinition';
import { objectMapToArray } from './utils';

/**
 * @internal
 */
export interface AddFilterProps {
    cancelAddFilter: () => void;

    availableFilters: PowerTable.FilterDefinition[] | { [key: string]: PowerTable.FilterDefinition };

    //availableFilters: { [key: string]: PowerTable.FilterDefinition };
    appliedFilters: PowerTable.AppliedFilter[];
    onApplyFilter: (filter: PowerTable.AppliedFilter) => void;

}

/**
 * @internal
 */
export interface AddFilterState {
    searchText: string;
    selectedFilterKey?: string;
}

/**
 * @internal
 */
export class AddSelectFilter extends React.PureComponent<AddFilterProps, AddFilterState> {

    constructor(props: AddFilterProps) {
        super(props);
        this.state = { searchText: '' };
        this.onSearchTextChanged = this.onSearchTextChanged.bind(this);
        this.backToPrev = this.backToPrev.bind(this);
        this.newFilterSelected = this.newFilterSelected.bind(this);

    }

    private onSearchTextChanged(e: React.FormEvent<FormControl & HTMLInputElement>) {
        this.setState({
            searchText: e.currentTarget.value,
        });

    }

    private newFilterSelected(e: React.SyntheticEvent<HTMLAnchorElement>) {
        e.preventDefault();

        this.setState({
            selectedFilterKey: e.currentTarget.dataset.fieldName,
        });
    }

    private backToPrev(e: React.SyntheticEvent<HTMLAnchorElement>) {
        e.preventDefault();

        if (this.state.selectedFilterKey) {
            this.setState({ selectedFilterKey: null });
        } else {
            this.props.cancelAddFilter();
        }
    }
    render() {

        const { availableFilters, appliedFilters } = this.props;

        const { searchText, selectedFilterKey } = this.state;

        if (selectedFilterKey) {

            const filter = Array.isArray(availableFilters) ? availableFilters.find((f) => f.fieldName === selectedFilterKey) : availableFilters[selectedFilterKey];

            const initialOperation = objectMapToArray(filter.operations)[0];

            return (
                <div>
                    <BackLink onClick={this.backToPrev} />

                    <div>

                        <AddEditFilter filter={filter} initialOperation={initialOperation} onApplyFilter={this.props.onApplyFilter} />

                    </div>
                </div>
            );

        }

        const filters = objectMapToArray(availableFilters);

        const unusedFilters = filters.filter((m) => appliedFilters.every((c) => c.filter.fieldName !== m.fieldName));

        const regex = new RegExp(searchText, 'i');
        const showFilters = searchText ? unusedFilters.filter((m) => m.displayName.match(regex)) : unusedFilters;

        // const onSelectFilter = (e: React.) => {
        //     e.preventDefault();
        //     this.newFilterSelected(m.fieldName);
        // };

        return (
            <div className="flex-column">
                <BackLink onClick={this.backToPrev} />

                <div><FormControl placeholder="Filter by" value={searchText} onChange={this.onSearchTextChanged} autoFocus /></div>

                <div className="small flex-column">
                    <div style={{ margin: '10px 0' }}><b>Available Filters</b></div>
                    <div className="available-filters">
                        <div className="list-group">
                            {showFilters.map((m) => <a href="#" onClick={this.newFilterSelected} data-fieldName={m.fieldName} className="list-group-item" key={m.fieldName}>{m.displayName}</a>)}
                        </div>
                    </div>

                </div>

            </div>
        );
    }
}
