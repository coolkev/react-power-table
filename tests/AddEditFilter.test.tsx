import * as React from 'react';
import { GridFilters, DataTypes, AppliedFilter, FilterDefinition } from '../src/';
import { defaultColumns, sampledata, President, partyList } from "./shared";
import { mount } from 'enzyme';
import { AddEditFilter } from "../src/AddEditFilter";

//const columns = defaultColumns;
const rows = sampledata.slice(0, 25);

const availableFilters = [
    new DataTypes.int('number'),
    new DataTypes.string('president'),
    new DataTypes.int('birth_year'),
    new DataTypes.int({ fieldName: 'death_year', canBeNull: true }),
    new DataTypes.date({ fieldName: 'took_office' }),
    new DataTypes.date({ fieldName: 'left_office', canBeNull: true }),
    new DataTypes.list('party', partyList),
    new DataTypes.boolean({ fieldName: 'assasinated', displayName: 'was assasinated' }),
    new DataTypes.decimal('empty'),
];


const testSerializeValues = [
    1,
    'test',
    1790,
    1850,
    '1900-02-03',
    '1900-02-03',
    'Republican',
    true,
    1.03
];


describe('AddEditFilter tests',
    () => {


        availableFilters.forEach((filter: FilterDefinition<any>,i) => {
            test('add filter', () => {

                const op = filter.operations[Object.keys(filter.operations)[0]];                
                const c = <AddEditFilter filter={filter} initialOperation={op} initialValue="" onApplyFilter={_e => { }} />;

                const component = mount(c);

                expect(component.render()).toMatchSnapshot('AddEditFilter tests add filter ' + filter.fieldName);


                const testValue = testSerializeValues[i];

                expect(typeof(filter.serializeValue(testValue))).toBe('string');

                filter.applyFilter(sampledata, op, testValue);
                
            });

        })



    });
