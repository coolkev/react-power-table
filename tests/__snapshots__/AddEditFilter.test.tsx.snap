// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddEditFilter tests add filter assasinated 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      was assasinated
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        was assasinated
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="ne"
        />
        was not assasinated
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter birth_year 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      birth_year
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        is equal to
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <span
        class="has-error"
      >
        <input
          class="form-control input-sm"
          type="number"
          value=""
        />
      </span>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="ne"
        />
        is not equal to
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="lt"
        />
        is less than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="gt"
        />
        is greater than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="between"
        />
        is between
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter death_year 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      death_year
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        is equal to
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <span
        class="has-error"
      >
        <input
          class="form-control input-sm"
          type="number"
          value=""
        />
      </span>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="ne"
        />
        is not equal to
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="lt"
        />
        is less than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="gt"
        />
        is greater than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="between"
        />
        is between
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="notnull"
        />
        has a value
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="null"
        />
        does not have a value
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter empty 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      empty
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        is equal to
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <span
        class="has-error"
      >
        <input
          class="form-control input-sm"
          type="number"
          value=""
        />
      </span>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="ne"
        />
        is not equal to
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="lt"
        />
        is less than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="gt"
        />
        is greater than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="between"
        />
        is between
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter left_office 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      left_office
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        is equal to
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <span
        class=""
      >
        <input
          autocomplete="on"
          class="form-control"
          placeholder="MM/DD/YYYY"
          style="width: 100%;"
          type="text"
          value=""
        />
        <div
          style="position: relative;"
        />
        <input
          data-formattedvalue=""
          type="hidden"
          value=""
        />
      </span>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="lt"
        />
        is before
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="gt"
        />
        is after
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="between"
        />
        is between
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="notnull"
        />
        has a value
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="null"
        />
        does not have a value
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter number 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      number
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        is equal to
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <span
        class="has-error"
      >
        <input
          class="form-control input-sm"
          type="number"
          value=""
        />
      </span>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="ne"
        />
        is not equal to
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="lt"
        />
        is less than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="gt"
        />
        is greater than
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="between"
        />
        is between
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter party 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      party
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="in"
        />
        is any of
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <div
        class="Select small Select--multi is-searchable"
      >
        <div
          class="Select-control"
        >
          <span
            class="Select-multi-value-wrapper"
            id="react-select-2--value"
          >
            <div
              class="Select-placeholder"
            >
              Select...
            </div>
            <div
              class="Select-input"
              style="display: inline-block;"
            >
              <input
                aria-activedescendant="react-select-2--value"
                aria-expanded="false"
                aria-haspopup="false"
                aria-owns=""
                role="combobox"
                style="width: 5px; box-sizing: content-box;"
                value=""
              />
              <div
                style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre;"
              />
            </div>
          </span>
          <span
            class="Select-arrow-zone"
          >
            <span
              class="Select-arrow"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter president 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      president
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="contains"
        />
        contains
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <input
        class="form-control"
        value=""
      />
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="notcontains"
        />
        does not contain
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        is equal to
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="ne"
        />
        is not equal to
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;

exports[`AddEditFilter tests add filter took_office 1`] = `
<div>
  <div
    style="margin: 10px 0px;"
  >
    <b>
      took_office
    </b>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="eq"
        />
        is equal to
      </label>
    </div>
    <div
      style="margin-left: 20px;"
    >
      <span
        class=""
      >
        <input
          autocomplete="on"
          class="form-control"
          placeholder="MM/DD/YYYY"
          style="width: 100%;"
          type="text"
          value=""
        />
        <div
          style="position: relative;"
        />
        <input
          data-formattedvalue=""
          type="hidden"
          value=""
        />
      </span>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="lt"
        />
        is before
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="gt"
        />
        is after
      </label>
    </div>
  </div>
  <div>
    <div
      class="radio"
    >
      <label>
        <input
          type="radio"
          value="between"
        />
        is between
      </label>
    </div>
  </div>
  <div
    style="margin-top: 20px;"
  >
    <button
      class="btn btn-sm btn-primary"
      type="button"
    >
      Apply Filter
    </button>
  </div>
</div>
`;
